tests/unit/auth/v2/queryAuthCheck.js

```diff
@@ -58,20 +60,50 @@ describe('v2: queryAuthCheck', () => {
 
     clock.tick(604800000);  // take time 604800000ms (7 days) ahead
 
-    it('URL should expire after 7 days with default preSignUrlExpiry', () => {
+    it('URL should expire after 7 days with default expiry', () => {
         const currentTime = Date.now();
         const request = { method: 'GET', query: { Expires: currentTime } };
         const data = { Expires: currentTime };
         const res = queryAuthCheck(request, log, data);
         assert.notStrictEqual(res.err, undefined);
         assert.strictEqual(res.err.AccessDenied, true);
     });
-    it('should raise MissingSecurityHeader with invalid expires param', () => {
-        const request = { method: 'GET', query: { Expires: 'a string' } };
-        const data = { Expires: 'a string' };
+
+    clock.restore();
+
+    it(`URL should not expire before 7 days with custom expiry`, () => {
+        process.env.PRE_SIGN_URL_EXPIRY = 31556952000; // in ms (1 year)
+        const currentTime = Date.now() / 1000;
+        const expires = currentTime + 604799;  // in seconds
+        const mockRequest = {
+            method: 'GET',
+            url: 'mockurl',
+            query: {
+                Expires: expires,
+            },
+            headers: {
+                'Content-MD5': 'c',
+            },
+        };
+        const data = {
+            Expires: expires,
+            AWSAccessKeyId: 'keyId',
+            Signature: 'sign',
+        };
+        const res = queryAuthCheck(mockRequest, log, data);
+        assert.notStrictEqual(res.err.AccessDenied, true);
+        assert.notStrictEqual(res.err.RequestTimeTooSkewed, true);
+    });
+
+    clock.tick(604800000);  // take time 604800000ms (7 days) ahead
```

Comment:
I would put this logic inside the tests that need it

Reply:
good point, done